components/nft-sale-widget/layout/mint-section.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useEffect, useState, useRef, useMemo } from "react";
+import { Box, Flex, Grid } from "@chakra-ui/react";
+import { useQuery } from "@apollo/client";
+import { getAllSalesWithTokens } from "@/services/graph";
+import { type GetAllSalesWithTokensData } from "@/services/types";
+import PsycItem from "../psyc-item";
+import useRandomImage from "@/hooks/useRandomImage";
+
+interface MintSectionProps {
+  isRandom: boolean;
+}
+
+export interface TokenItem {
+  src: string;
+  price: string;
+  isSold: boolean;
+  batchId: string;
+  tokenId: string;
+}
+
+const images = ["/psyc1.png", "/psyc2.png", "/psyc3.png", "/psyc4.png"];
+
+const MintSection = ({ isRandom }: MintSectionProps) => {
+  const { loading, error, data } = useQuery<GetAllSalesWithTokensData>(
+    getAllSalesWithTokens
+  );
+  const currentImageIndex = useRandomImage(isRandom, images);
+  const tokensRef = useRef<TokenItem[]>([]);
+  const [randomToken, setRandomToken] = useState<TokenItem | null>(null);
+
+  const tokens = useMemo(() => {
+    return (
+      data?.sales.flatMap((sale) =>
+        sale.tokensOnSale.map((token) => ({
+          src: images[currentImageIndex % images.length] ?? "",
+          price: `${parseFloat(sale.floorPrice) / 10 ** 18} ETH`,
```

Comment:
lets not use this 10**18, there is a function to convert to price. Lets use that instead

Reply:
alright